x-pack/plugins/observability_solution/synthetics/server/synthetics_service/formatters/formatting_utils.ts

```diff
@@ -51,7 +51,7 @@ export const replaceStringWithParams = (
 };
 
 export const hasNoParams = (strVal: string) => {
-  const shellParamsRegex = /\$\{[a-zA-Z_][a-zA-Z0-9_]*\}/g;
+  const shellParamsRegex = /\$\{[a-zA-Z_][a-zA-Z0-9_-]*\}/g;
```

Comment:
nit: why not keep it simple as `\w+|[-_]`? 

Are we missing any other substution other than _ and - for parameters. Are dots used in beats config? There seems to be `:, .` used as well - https://www.elastic.co/guide/en/beats/libbeat/current/config-gile-format-refs.html

Reply:
@vigneshshanmugam looking at [the docs](https://www.elastic.co/guide/en/beats/libbeat/current/config-file-format-env-vars.html#_examples) it appears that `?` is also possible.

Reply:
@vigneshshanmugam the `:` case is already covered by what we have today, you can see a [a unit test](https://github.com/elastic/kibana/blob/d05738030c4a83ead2a6ea0fb60c13ca0de8403a/x-pack/plugins/observability_solution/synthetics/server/synthetics_service/formatters/formatting_utils.test.ts#L61) to this effect in the same file.

Regarding the `?`, it pertains to giving a custom error message for beats to output. I don't know if this is a feature we support or not, but doesn't seem super important here.

`\w+|[-_]` is not a drop-in fix, several of our covered usages fail whereas what I already added works for all of them. Frankly, I do not enjoy fiddling with regex so I am inclined to keep it as it is, to my eye it doesn't really reduce the overall complexity of the code much. I'm happy to put in something cleaner if all the tests keep passing though 😃.

I did add a `\.` to the character class to indicate the intention that we will support the `.` character per the case you linked above; I also included a new unit test. LMK if you have any further thoughts, thanks for the feedback 👍.

Reply:
> case is already covered by what we have today, 

That is for substituion I beleieve, what if we set the key to `test:param` in the global params? If it works all good. 

I am not sure if params would end up including any characters other than `- and _`. Just wanted to call out. I am fine with either way. Thanks for taking a look.